analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -71,121 +71,231 @@
+  -- Footway can lead to anything (attraction entrances, ...) so exclude them even if oneway
+  highway IN ('motorway', 'raceway', 'escape', 'footway') OR
+  -- Construction roads are usually temporary, the connections are likely access=no or similar
+  is_construction
+UNION ALL
+-- Include nodes of ways that cross the extract border. See #1949
+-- Better would be to only include the nodes outside of the extract, but that's expensive
+SELECT
+  unnest(borderways.nodes)
+FROM
+  relation_members AS boundary_members
+  JOIN ways AS boundary_ways ON
+    boundary_members.member_id = boundary_ways.id
+  JOIN highways AS borderways ON
+    ST_Intersects(boundary_ways.linestring, borderways.linestring)
+WHERE
+  boundary_members.member_type = 'W' AND
+  boundary_members.relation_id IN {boundary_ids}
+"""
+
+sql33 = """
+CREATE INDEX idx_allowed_nodes_id ON allowed_end_nodes(id);
+"""
+
+sql34 = """
+-- Make table as short as possible: keep only nodes that are actually nodes of oneways
+CREATE TEMP TABLE allowed_oneway_end_nodes AS
+SELECT DISTINCT
+  allowed_end_nodes.id
+FROM
+  allowed_end_nodes
+  JOIN oneway_highway ON
+    -- Note that we cannot literally only get the first and last node, as i.e. in a looped oneway the 'oneway-entrance/exit node' might be in the middle
+    allowed_end_nodes.id = ANY(oneway_highway.nodes)
+"""
+
+sql35 = """
+CREATE INDEX idx_allowed_oneway_nodes_id ON allowed_oneway_end_nodes(id);
+DROP TABLE allowed_end_nodes;
+"""
+
+sql36 = """
+-- All end nodes of oneway streets
+CREATE TEMP TABLE oneway_terminal AS
+SELECT DISTINCT ON (nid)
+  oneway.id AS wid,
+  terminal.id AS nid,
+  terminal.geom AS geom,
+  CASE
+    WHEN oneway_reverse THEN terminal.id = oneway.nodes[1]
+    ELSE terminal.id = oneway.nodes[oneway.length]
+  END AS is_oneway_last_node
+FROM
+  oneway_highway AS oneway
+  JOIN nodes AS terminal ON
+    (
+      terminal.id = oneway.nodes[1] OR
+      terminal.id = oneway.nodes[oneway.length]
+    )
+  LEFT JOIN allowed_oneway_end_nodes ON
+    terminal.id = allowed_oneway_end_nodes.id
+WHERE
+  allowed_oneway_end_nodes IS NULL AND
+  array_length(array_positions(oneway.nodes, terminal.id), 1) = 1 -- exclude end nodes folding back into the same way, e.g. in P or O-shaped ways
+ORDER BY
+  terminal.id,
+  oneway.id
+"""
+
+sql37 = """
+-- Find all end nodes that are not connected to a way that allows to enter/leave the oneway street
+CREATE TEMP TABLE results_straightforward AS
+SELECT
+  wid,
+  nid,
+  geom
+FROM
+  oneway_terminal
+  LEFT JOIN oneway_highway AS other_highway ON
+    oneway_terminal.nid = ANY(other_highway.nodes) AND
+    oneway_terminal.geom && other_highway.linestring AND -- much faster than comparison with nid
+    oneway_terminal.wid != other_highway.id AND
+    (
+      array_length(array_positions(other_highway.nodes, oneway_terminal.nid), 1) != 1 OR -- end node of loop/self-intersecting way that also occurs elsewhere
+      (
+        other_highway.oneway_reverse AND
```

Comment:
Maybe you just can reverse the nodes in `oneway_highway` rather than use `oneway_reverse`?

Reply:
Good suggestion, implemented (and much simpler indeed)